### ISSUE TITLE
OpenMW support

### ISSUE THREAD

**Issue by EmperorArthur** [mlox/mlox#1]
Originally by Translator5.  Transferred when the project moved.

So it there a possibility to make python version sort the OpenMW config?
Or is this more complicated because of different Linux distributions and different install locations?

Linux Mint 18 64 bit
OpenMW 0.40
Morrowind GOTY

How to sort mods for OpenMW on Linux
-> Run Morrowind Launcher.exe using wine
-> Enable your mods there
-> run mlox using wine or the python version
-> run OpenMW launcher & import the morrowind.ini settings & loadorder

---

If you want to look at it, here's a possible starting point: https://github.com/mlox/mlox/blob/master/mlox/mlox.py#L1079

----

Side note, if you use the mlox-working branch you don't need "Morrowind Launcher.exe"  It's not the easiest, but it works

Your directory structure should have:
* Morrowind.ini
* "Data Files/"
* "mlox/"

Here's what to do:
1. Make sure `[game files]` in morrowind.ini has no entries
2. Run mlox, and save the resulting load order.
3. Open the file with the new load order, and add "content=" to every line (search and replace is great for this)
4. Copy those lines to `openmw.cfg`

----

This is much more stuff to do 😆 - I think I will use wine instead

**Comment by TheMechanist902** [comment]
Yes this would be a very useful feature! I would really like to see support for OpenMW :)

**Comment by EmperorArthur** [comment]
Can someone please post an openmw.cfg with at least one active plugin.

I can access the system with my OpenMW install on it right now.

**Comment by Dragon-32** [comment]
Here's one from  Amenophis on Bethsoft

[openmw.txt](https://github.com/mlox/mlox/files/849050/openmw.txt)

**Comment by milos-popovic** [comment]
Any news on this?

**Comment by milos-popovic** [comment]
Sorry for the above question. I just tried to load my openmw.cfg and it worked. You might want to close this issue though. It might confuse someone to assume it's not implemented yet as the issue is in the 'Open' state.

**Comment by EmperorArthur** [comment]
Hi,

Did you use the command line?  If so, what options did you use?  What operating system are you using?

While File read support had been implemented for a while now, you may be surprised by attempting a write operation, as the file writer is designed to handle Morrowind.ini files.

We currently don't support opening a random file, and reading from a data directory.  As such, warnings based on size will now be shown, and Mlox will not scrub your configuration file of any missing plugins.

Openmw support will be completed when a user can open the GUI, and can easily choose to use the openmw.cfg file with their data directory.

I will, however, add milestones so everyone can track the progress to that goal.

**Comment by milos-popovic** [comment]
Hi,

I'm using Windows, and I've loaded the config using UI, not command line. I do agree that this issue is not done to the level of quality that should be expected, as saving an openmw.cfg doesn't work. However, adding milestones would be great, as I almost didn't try to load the config at all, and I can imagine many people wouldn't.

Thanks for the quick reply!

**Comment by Danjb1** [comment]
Can someone please clarify the current status of this?

I can see in the changelog for Version 0.62:

> mlox now understands OpenMW config files format, though it does not (yet) actively find them. [EmperorArthur]

However when I try to load my `openmw.cfg` file using the latest mlox (v1.0.3) I get a bunch of error messages:

> ERROR: Unable to open configuration file: /D:/Danjb/Documents/My Games/OpenMW/openmw.cfg
> WARNING: No plugins detected.
> mlox understands lists of plugins in the format used by Morrowind.ini or Wrye Mash.
> Is that what you used for input?
> Found 0 plugins in: "/D:/Danjb/Documents/My Games/OpenMW/openmw.cfg"
> (Note: When the load order input is from an external source, the [SIZE] predicate cannot check the plugin filesizes, so it defaults to True).
> ERROR: No plugins detected!
> mlox needs to run somewhere under where the game is installed.

I would really love to get this working with OpenMW.

**Comment by shumafuk** [comment]
Please, make support for OpenMW. I use separate folders for OpenMW mods that are linked in openmw.cfg, so I'm not able to get data from morrowind.ini for plugin load order.
